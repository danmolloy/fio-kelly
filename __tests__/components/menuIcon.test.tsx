import "@testing-library/jest-dom";
import { act, fireEvent, render, screen } from "@testing-library/react";
import MenuIcon, { MenuIconProps } from "../../components/menuIcon";

const mockProps: MenuIconProps = {
  setShowMenu: jest.fn(),
  showMenu: true,
};

describe("<MenuIcon />", () => {
  beforeEach(() => {
    render(<MenuIcon {...mockProps} />);
  });
  it("menu-icon is in the document", () => {
    const menuIcon = screen.getByTestId("menu-icon");
    expect(menuIcon).toBeInTheDocument();
  });
  it("calls setShowMenu on click", () => {
    const menuIcon = screen.getByTestId("menu-icon");
    act(() => {
      fireEvent.click(menuIcon);
    });
    expect(mockProps.setShowMenu).toHaveBeenCalled();
  });
  it("matches snapshot", () => {
    const menuIcon = screen.getByTestId("menu-icon");
    expect(menuIcon).toMatchSnapshot();
  });
});
