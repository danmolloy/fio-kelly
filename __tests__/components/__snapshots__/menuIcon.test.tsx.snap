// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuIcon /> matches snapshot 1`] = `
<button
  class=" m-2 hover:fill-amber-500 active:fill-amber-400  md:hidden"
  data-testid="menu-icon"
>
  <svg
    class="m-2"
    height="22"
    width="30"
  >
    <rect
      height="2"
      rx="2"
      style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
      width="30"
    />
    <rect
      height="2"
      rx="2"
      style="opacity: 1;"
      width="30"
      y="8"
    />
    <rect
      height="2"
      rx="2"
      style="transform: scaleX(1) translate(0px, 0px) rotate(0deg);"
      width="30"
      y="17"
    />
  </svg>
  <p
    class="hidden"
  >
    Menu
  </p>
</button>
`;
